### DISCUSSION TITLE
library overlay container duplicated (v3)

### DISCUSSION THREAD

**Discussion by SoftWebMike** [vuetifyjs/vuetify#15949] [Q&A]
We have two vue repositories, both using vuetify (v3) beta:
1. A library containing common components - "library"
2. An application that uses (1) - "app"

What we are seeing is that when app initializes, it adds two DOM elements, both with the class "v-overlay-container". When we invoke a modal from the library that exposes a slot, it loads to overlay container A. Within that slot, we define an autocomplete within the app.

What we are seeing, is that when we interact with the autocomplete, the autocomplete options are rendered behind the modal. When we inspect the DOM, we can see that the autocomplete is not using the same overlay container as the modal, and thus the z-index is not being resolved.

Further investigation suggests that all components with overlay requirements within the library are loaded to the container, and those from the app are loaded to container B.

The questions here are:
1. Is there any setup that needs to be done to build components that use Vuetify within a library, which takes this behavior into account
2. Have we misconfigured the library in some way that would explain this behavior?

**Comment by KaelWD** [comment]
Pretty old answer but the concept is still the same: https://github.com/vuetifyjs/vuetify/discussions/4068#discussioncomment-24984

**Comment by valoricDe** [comment]
How does the answer change with the existence of createVuetify where there is no global instance? The library should reuse the app vuetify right?
I saw it is possible to define (<v-dialog attach="#containerB") which container should be used. But I think it would not necessarily change the above behaviour.
